bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/upnp/UpnpServer.java

```diff
@@ -409,48 +409,47 @@ private void sendUPNPNotify(DatagramSocket sendSocket, InetAddress address, int
     @Override
     public void accept(HueEmulationConfigWithRuntime threadContext) {
         logger.info("Hue Emulation UPNP server started on {}:{}", threadContext.addressString, threadContext.port);
-        boolean hasIPv4 = false;
-        boolean hasIPv6 = false;
 
-        try (
-                DatagramChannel channelV4 = DatagramChannel.open(StandardProtocolFamily.INET);
-                DatagramChannel channelV6 = DatagramChannel.open(StandardProtocolFamily.INET6);
-                Selector selector = Selector.open()) {
+        try (Selector selector = Selector.open()) {
```

Comment:
I understand why you removed the channels here, but maybe you could add a `finally` clause and free the resources there.

Reply:
Yeah it needs some fixing - there seems to be some multithreading going on too by looking at the failed Travis test.

Reply:
@J-N-K  I rewrote the change, hope you're ok with me edit/force pushing in my branch, it's a small commit..

Reply:
@J-N-K Are you fine with the current use of try-resources? Would merge if no objections